### PR TITLE
Implement card BIN functionality

### PR DESCRIPTION
**Pull request by julietteceb16**
## Summary by CodeRabbit

* **New Features**
  * Bank-name lookup now accepts card numbers (BIN-based) in addition to CLABE; a strict variant throws on invalid or unknown values.
  * A public card-BIN lookup table is now exposed.

* **Tests**
  * Expanded tests for card/BIN lookups, CLABE lookups, edge cases, and error conditions.

* **Chores**
  * Package version bumped to 1.4.0.

[Review Change Stack](https://app.coderabbit.ai/change-stack/cuenca-mx/clabe-js/pull/29)

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -35,33 +36,71 @@ function validateClabe(clabe) {
...
+function getBankNameByBin(cardNumber) {
+  /*
+  Devuelve el nombre del banco basado en los primeros 6 dígitos (BIN) de una tarjeta.
+  Devuelve null si la entrada es inválida o si el BIN no está registrado.
+  */
+  if (typeof cardNumber !== 'string' || cardNumber.length < 6) return null;
+  const bin = cardNumber.substring(0, 6);
+  const bankName = BANK_NAMES[CARD_BINS[bin]];
   return bankName === undefined ? null : bankName;
 }
```
Comment (gmorales96) [inline lib/index.js]: Veo que `getBankName` ya resuelve el nombre del banco tanto por CLABE como por tarjeta. `getBankNameByBin` parece duplicar comportamiento, ya que únicamente cubre el caso de tarjeta. ¿Hay algún caso específico que justifique mantener ambas funciones?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "clabe-js",
-  "version": "1.3.0",
+  "version": "1.4.0-dev.0",
```
Comment (gmorales96) [inline package.json]: versión final

**Review by gmorales96 — APPROVED** [review]
